lib/resty/limit/conn.lua

```diff
@@ -47,58 +47,38 @@ function _M.incoming(self, key, commit)
     local max = self.max
 
     self.committed = false
+    if commit then
+        local conn, err = dict:incr(key, 1, 0)
+        if not conn then
+            return nil, err
+        end
 
-    local conn = dict:get(key)
-    if conn then
-        if conn >= max then
-            if conn >= max + self.burst then
-                return nil, "rejected"
-            end
-
-            if commit then
-                local err
-                conn, err = dict:incr(key, 1, 0)
-                if not conn then
-                    return nil, err
-                end
-
-                self.committed = true
-            else
-                conn = conn + 1
-            end
+        if conn > max + self.burst then
+            dict:incr(key, -1, conn)
+            return nil, "rejected"
+        end
+        self.committed = true
 
+        if conn > max then
             -- make the exessive connections wait
             return self.unit_delay * floor((conn - 1) / max), conn
         end
 
-        if commit then
-            -- FIXME: we should use incr_or_init here.
-            local new_conn, err = dict:incr(key, 1)
-            if not new_conn then
-                return nil, err
+        -- we return a 0 delay by default
+        return 0, conn
+    else
```

Comment:
We can omit this `else` here since the previous branch always returns and never falls through.